autogpts/autogpt/autogpt/models/command_registry.py

```diff
@@ -1,212 +0,0 @@
-            self.commands_aliases[alias] = cmd
-
-    def unregister(self, command: Command) -> None:
-        if command.name in self.commands:
-            del self.commands[command.name]
-            for alias in command.aliases:
-                del self.commands_aliases[alias]
-        else:
-            raise KeyError(f"Command '{command.name}' not found in registry.")
-
-    def reload_commands(self) -> None:
-        """Reloads all loaded command plugins."""
-        for cmd_name in self.commands:
-            cmd = self.commands[cmd_name]
-            module = self._import_module(cmd.__module__)
-            reloaded_module = self._reload_module(module)
-            if hasattr(reloaded_module, "register"):
-                reloaded_module.register(self)
-
-    def get_command(self, name: str) -> Command | None:
-        if name in self.commands:
-            return self.commands[name]
-
-        if name in self.commands_aliases:
-            return self.commands_aliases[name]
-
-    def call(self, command_name: str, agent: BaseAgent, **kwargs) -> Any:
-        if command := self.get_command(command_name):
-            return command(**kwargs, agent=agent)
-        raise KeyError(f"Command '{command_name}' not found in registry")
-
-    def list_available_commands(self, agent: BaseAgent) -> Iterator[Command]:
-        """Iterates over all registered commands and yields those that are available.
-
-        Params:
-            agent (BaseAgent): The agent that the commands will be checked against.
-
-        Yields:
-            Command: The next available command.
-        """
-
-        for cmd in self.commands.values():
-            available = cmd.available
-            if callable(cmd.available):
-                available = cmd.available(agent)
-            if available:
-                yield cmd
-
-    # def command_specs(self) -> str:
-    #     """
-    #     Returns a technical declaration of all commands in the registry,
-    #     for use in a prompt.
-    #     """
-    #
-    #     Declaring functions or commands should be done in a model-specific way to
-    #     achieve optimal results. For this reason, it should NOT be implemented here,
-    #     but in an LLM provider module.
-    #     MUST take command AVAILABILITY into account.
-
-    @staticmethod
-    def with_command_modules(modules: list[str], config: Config) -> CommandRegistry:
-        new_registry = CommandRegistry()
-
-        logger.debug(
-            "The following command categories are disabled: "
-            f"{config.disabled_command_categories}"
-        )
-        enabled_command_modules = [
-            x for x in modules if x not in config.disabled_command_categories
-        ]
-
-        logger.debug(
-            f"The following command categories are enabled: {enabled_command_modules}"
-        )
-
-        for command_module in enabled_command_modules:
-            new_registry.import_command_module(command_module)
```

Comment:
Is this configurability ported to the new system?

Reply:
Command categories? No exactly, commands are now in the relevant components, so components need to be disabled and if we need more granular control we need component-level configuration. I think we wanted to move away from the legacy config and many env vars.

Reply:
The possibility to disable commands is currently used in production though, e.g. to disable `ask_user` during automated runs. This needs to be addressed before merging.

Reply:
`ask_user` specifically will be disabled if `noninteractive_mode is True`. Other than that I'd rather patch this for now until we decide how the component config should look like.
I can introduce `DISABLED_COMMANDS` env var and a method or protocol+component to remove disabled commands from the list before sending them to llm.

Reply:
https://github.com/Significant-Gravitas/AutoGPT/pull/7054/commits/7b924b23efe51d63dd99599486a10a45564d5bbc